src/rules/quotePropsRule.ts

```diff
@@ -0,0 +1,103 @@
+import * as Lint from "../lint";
+import * as ts from "typescript";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "quote-props",
+        description: "Quoting Style for Property Names",
+        descriptionDetails: Lint.Utils.dedent`
+            Object literal property names can be defined in two ways: using literals or using strings.
+            For example, these two objects are equivalent:
+
+            var object1 = {
+                property: true
+            };
+
+            var object2 = {
+                "property": true
+            };
+
+            In many cases, it doesn’t matter if you choose to use an identifier instead of a string
+            or vice-versa. Even so, you might decide to enforce a consistent style in your code.
+
+            This rules lets you enforce consistent quoting of property names. Either they should always
+            be quoted (default behavior) or quoted only as needed ("as-needed").
+        `,
```

Comment:
I'm not sure it matters, but I normally put this closing "`," up one line in order to prevent the extra newline.